devito/passes/equations/linearity.py

```diff
@@ -52,7 +52,19 @@ def _(c, deriv):
 
 @_is_const_coeff.register(sympy.Function)
 def _(c, deriv):
-    return not set(deriv.dims) & c.free_symbols
+    is_const = not set(deriv.dims) & c.free_symbols
+    try:
```

Comment:
Probably you should use `d._defines` rather than the sole parent?

Then _maybe_ you could rewrite everything simply as
```
c_dims = set().union(*[i._defines for i in c.free_symbols if isinstance(i, Dimension)])
deriv_dims = set().union(*[d._defines for d in deriv.dims])
is_const = not c_dims & deriv_dims
```

? 

not sure if I'm missing something though

Reply:
yeah that works too, is replaced the isinstance by a getaatr on `defines` to avoid importing `Dimension` inside  `is_const` but works as expected.